File: frontend/src/interface/Package.ts
export interface Package {
    id?: number;
    ID?: number;
    p_name: string;
    type: string;
    detail: string;
    service_id: number;
    price: number;
    service?: Service;
    detail_service?: Service;
    created_at?: string;
    updated_at?: string;
  }
  
  export interface Service {
    id: number;
    service: string;
    detail: string;
    created_at?: string;
    updated_at?: string;
  }
  

File: Backend/entity/package.go
package entity

import "gorm.io/gorm"

type Package struct {
	gorm.Model
	ID            uint      `json:"id"`
	PackageName   string    `json:"p_name"`
	Type          string    `json:"type"`
	Detail        string    `json:"detail"`
	ServiceID     uint      `json:"service_id"`
	Service       *Services `gorm:"foreignKey:ServiceID" json:"service"`
	DetailService *Services `gorm:"foreignKey:ServiceID" json:"detail_service"`
	Price         uint      `json:"price"`
}
